test/Microsoft.AspNet.Mvc.Core.Test/ControllerTests.cs

```diff
@@ -424,6 +424,34 @@ public void Created_WithRelativeUriParameter_SetsCreatedLocation()
         }
 
         [Fact]
+        public void Created_IDisposableObject_RegistersForDispose()
+        {
+            // Arrange
+            var mockHttpContext = new Mock<DefaultHttpContext>();
+            mockHttpContext.Setup(x => x.Response.OnResponseCompleted(It.IsAny<Action<object>>(), It.IsAny<object>()))
+                .Verifiable();
+            var uri = new Uri("/test/url", UriKind.Relative);
+
+            var controller = new TestableController()
+            {
+                ActionContext = new ActionContext(mockHttpContext.Object, new RouteData(), new ActionDescriptor())
+            };
+            var input = new DisposableObject();
+
+            // Act
+            var result = controller.Created(uri, input);
+
+            // Assert
+            Assert.IsType<CreatedResult>(result);
+            Assert.Equal(StatusCodes.Status201Created, result.StatusCode);
+            Assert.Equal(uri.OriginalString, result.Location);
+            Assert.Same(input, result.Value);
+            mockHttpContext.Verify(
```

Comment:
You have to use this overload if you want to verify the number of times it was called. In which case, it's probably less verbose to remove the setup on L#431-L#432.